format/FormatHDF5.py

```diff
@@ -18,7 +19,12 @@ def __init__(self, image_file, **kwargs):
     def understand(image_file):
         try:
             with FormatHDF5.open_file(image_file, "rb") as fh:
-                return fh.read(8) == b"\211HDF\r\n\032\n"
+                if fh.read(8) != b"\211HDF\r\n\032\n":
+                    return False
+            if hdf5_file_type(image_file) == HDF5_NXS_FILE:
+                return True
+            else:
+                return False
         except IOError:
```

Comment:
If we try to `h5py.File`-open then we don't need to do a separate signature check, as the HDF5 library will do that anyway.
```python
try:
    return get_hdf5_file_type(image_file) is HDF5_FileType.NXS
except (IOError, OSError):
    return False
```